sdk/resourcemanager/azure-resourcemanager-compute/src/main/java/com/azure/resourcemanager/compute/implementation/SnapshotImpl.java

```diff
@@ -130,7 +131,9 @@ public Mono<Void> awaitCopyStartCompletionAsync() {
                 String.format(
                     "\"awaitCopyStartCompletionAsync\" cannot be called on snapshot \"%s\" when \"creationMethod\" is not \"CopyStart\"", this.name()))));
         }
-        return getInnerAsync()
+
+        return Flux.interval(Duration.ZERO, ResourceManagerUtils.InternalRuntimeContext.getDelayDuration(manager().serviceClient().getDefaultPollInterval()))
```

Comment:
@weidongxu-microsoft, @XiaofeiCao, I changed the design here from using `delaySubscription` `repeat` to `Flux.interval`. This should work the same as the previous design but be clearer and more similar to what is trying to be achieve which is run a task on delay repeatedly until it completes or times out. This may also be slightly more performant as there should be less creation of worker resources as `Flux.interval` will only need to do this once while `delaySubscription` `repeat` would do this every poll.

Reply:
Thanks Alan, LGTM.